test/api/users.spec.js

```diff
@@ -0,0 +1,375 @@
+import AppManager from '../manager/app-manager';
+
+describe('Users', () => {
+  let appManager;
+  let httpClient;
+
+  beforeEach(async () => {
+    appManager = new AppManager();
+    await appManager.start();
+    httpClient = appManager.getHttpClient();
+  });
+
+  afterEach(async () => {
+    await appManager.stop();
+  });
+
+  it.each([
+    ['miffyliye', 'Miffy Liye', '2012-12-31', 'Xiangyang', 'DEV'],
+    ['wangtao', 'Wang Tao', '2015-01-01', 'Wuhan', 'PM'],
+  ])(
+    'should create user then get created user',
+    async (id, name, dob, address, description) => {
+      const authHeaders = await appManager.getAuthHeaders();
+
+      const createRes = await httpClient
+        .post('/users')
+        .set(authHeaders)
+        .send({
+          id,
+          name,
+          dob,
+          address,
+          description,
+        });
+
+      expect(createRes.status).toBe(201);
+      const getRes = await httpClient.get(`/users/${id}`).set(authHeaders);
+      expect(getRes.status).toBe(200);
+      const user = getRes.body;
+      expect(user.id).toBe(id);
+      expect(user.name).toBe(name);
+      expect(user.dob).toBe(dob);
+      expect(user.address).toBe(address);
+      expect(user.description).toBe(description);
+      expect(user.createdAt).toBeTruthy();
+    },
+  );
+
+  it.each([
+    ['miffyliye', 'Miffy Liye', 'Alex'],
+    ['wangtao', 'Wang Tao', 'Bob'],
+  ])(
+    'should not create user when id is used by existing user',
```

Comment:
If I understand correctly these tests are depending on the previous test result (https://github.com/Wiredcraft/test-backend/pull/41/files#diff-b192a391ed4193316fb05c0dfb46d100R25), right?
Do you see it as an issue?

Reply:
No, each test case runs independently. The database for test is dropped when the `AppManager` starts in `beforeEach`.

https://github.com/Wiredcraft/test-backend/pull/41/files#diff-b192a391ed4193316fb05c0dfb46d100R9

https://github.com/Wiredcraft/test-backend/pull/41/files#diff-b1654daea5b17178a70f3f91ce796f36R16